harmony binding bug

I trying since days to setup the harmony binding, and i'm unable to manage it. But now i think it is a bug.
I'm not really sure if it is a bug in the harmony bindung but in the core framework.
Nervertheless i found it here, so i thought it would be the right way to post it here.

All Things were found correctly the hub, and the devices. Nevertheless the the device does not have a pointer the bridge (hub). I tried it with paperUI and with HABmin. There is no way to select the bridge (hub). This leads to a nullpointerException

`2016-11-20 23:48:15.523 [ERROR] [ome.core.thing.internal.ThingManager] - Exception occured while calling handler: java.lang.NullPointerException
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)[:1.8.0_65]
	at java.util.concurrent.FutureTask.get(FutureTask.java:206)[:1.8.0_65]
	at org.eclipse.smarthome.core.common.SafeMethodCaller.callAsynchronous(SafeMethodCaller.java:186)[96:org.eclipse.smarthome.core:0.9.0.201611191809]
	at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:79)[96:org.eclipse.smarthome.core:0.9.0.201611191809]
	at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:63)[96:org.eclipse.smarthome.core:0.9.0.201611191809]
	at org.eclipse.smarthome.core.thing.internal.ThingManager.receiveCommand(ThingManager.java:349)[103:org.eclipse.smarthome.core.thing:0.9.0.201611191809]
	at org.eclipse.smarthome.core.items.events.AbstractItemEventSubscriber.receive(AbstractItemEventSubscriber.java:46)[96:org.eclipse.smarthome.core:0.9.0.201611191809]
	at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:192)[96:org.eclipse.smarthome.core:0.9.0.201611191809]
	at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:1)[96:org.eclipse.smarthome.core:0.9.0.201611191809]
	at org.eclipse.smarthome.core.common.SafeMethodCaller$CallableWrapper.call(SafeMethodCaller.java:177)[96:org.eclipse.smarthome.core:0.9.0.201611191809]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_65]
	at java.lang.Thread.run(Thread.java:745)[:1.8.0_65]
Caused by: java.lang.NullPointerException
	at org.openhab.binding.harmonyhub.handler.HarmonyDeviceHandler.handleCommand(HarmonyDeviceHandler.java:90)[196:org.openhab.binding.harmonyhub:2.0.0.201611201812]
	at org.eclipse.smarthome.core.thing.internal.ThingManager$3.call(ThingManager.java:352)[103:org.eclipse.smarthome.core.thing:0.9.0.201611191809]
	at org.eclipse.smarthome.core.thing.internal.ThingManager$3.call(ThingManager.java:1)[103:org.eclipse.smarthome.core.thing:0.9.0.201611191809]
	... 5 more`

So i started it in the debugger and found out that the harmonyHub is not initialised in the device.

I'm talking about `HarmonyHubHandler bridge` in `HarmonyDeviceHandler.java`

But as ` <supported-bridge-type-refs>` is a core function i'm not really sure, if it is a bug in the binding or somewhere else.

--- a/addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyDeviceHandler.java
+++ b/addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyDeviceHandler.java
@@ -22,6 +22,7 @@ import org.eclipse.smarthome.core.thing.ChannelUID;
 import org.eclipse.smarthome.core.thing.Thing;
 import org.eclipse.smarthome.core.thing.ThingStatus;
 import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
 import org.eclipse.smarthome.core.thing.ThingTypeUID;
 import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
 import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
@@ -112,28 +113,18 @@ public class HarmonyDeviceHandler extends BaseThingHandler {
         } else {
             logName = id > 0 ? String.valueOf(id) : name;
             logger.debug("initializing {}", logName);
-            if (getBridge() != null) {
-                updateDeviceStatus(getBridge().getStatus());
-            }
+            updateBridgeStatus();
         }
     };
 
     @Override
-    public void dispose() {
-        factory.removeChannelTypesForThing(getThing().getUID());
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        updateBridgeStatus();
     }
 
-    /**
-     * updates our local status to online if our bridge is online and we
-     * have a valid configuration.
-     *
-     * @param status
-     */
-    private void updateDeviceStatus(ThingStatus status) {
-        if (checkConfig() && status.equals(ThingStatus.ONLINE)) {
-            updateStatus(ThingStatus.ONLINE);
-            updateChannel();
-        }
+    @Override
+    public void dispose() {
+        factory.removeChannelTypesForThing(getThing().getUID());
     }
 
     /**
@@ -146,15 +137,25 @@ public class HarmonyDeviceHandler extends BaseThingHandler {
     }
 
     /**
+     * Updates our state based on the bridge/hub
+     */
+    private void updateBridgeStatus() {
+        ThingStatus bridgeStatus = getBridge().getStatus();
+        if (bridgeStatus == ThingStatus.ONLINE && getThing().getStatus() != ThingStatus.ONLINE) {
+            bridge = (HarmonyHubHandler) getBridge().getHandler();
+            updateStatus(ThingStatus.ONLINE);
+            updateChannel();
+        } else if (bridgeStatus == ThingStatus.OFFLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    /**
      * Updates our channel with the available buttons as option states
      */
     private void updateChannel() {
         try {
             logger.debug("updateChannel for device {}", logName);
-            if (bridge == null) {
-                logger.debug("updateChannel: no bridge for device {}", logName);
-                return;
-            }
 
             HarmonyConfig config = bridge.getCachedConfig();
             if (config == null) {

